inc/functions/formatting.php

```diff
@@ -259,22 +259,29 @@ function rocket_remove_url_protocol( $url, $no_dots = false ) {
 }
 
 /**
- * Add HTTP protocol to an url that does not have
+ * Add HTTP protocol to an url that does not have.
  *
  * @since 2.2.1
  *
  * @param string $url The URL to parse.
- * @return string $url The URL with protocol
+ *
+ * @return string $url The URL with protocol.
  */
 function rocket_add_url_protocol( $url ) {
+	// Bail out if the URL starts with http:// or https://.
+	if (
+		strpos( $url, 'http://' ) !== false
+		||
+		strpos( $url, 'https://' ) !== false
+	) {
+		return $url;
```

Comment:
The URL should still go through `set_url_scheme()` to make sure it uses the correct scheme depending on the value of `is_ssl()`

Reply:
It doesn't currently. Currently, it only goes through `set_url_scheme()` to get the `http:` or `http:` protocol if the given URL doesn't have either.

Reply:
@Tabrisrp Are you thinking this is a bug?

Reply:
oh yes you're right, I read the original wrong, sorry about that!

I think it's ok that way in fact, looking at where it's used in our codebase, there is no issue with it.